lib/rubocop/cop/style/method_call_parentheses.rb

```diff
@@ -50,6 +46,13 @@ def same_name_assignment?(node)
               return false if asgn_node.send_type?
             end
 
+            if asgn_node.masgn_type?
+              mlhs_node, _mrhs_node = *asgn_node
+              vars = mlhs_node.children.map(&:source)
```

Comment:
This is extremely expensive (mapping to the source). Consider just using the symbol identifiers or something like this.